### PR TITLE
Add a changelog Reminder

### PR DESCRIPTION
**Pull request by albi3ro**
Adding [Changelog Reminder](https://github.com/marketplace/actions/changelog-reminder) to the workflow.  If a pull request does not edit `./github/CHANGELOG.md`, this GitHub action will spit out a reminder.

### CODE REVIEW

File: .github/workflows/changelog_reminder.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+on:
+  pull_request:
+    types: [opened, ready_for_review]
```
Comment (josh146) [inline .github/workflows/changelog_reminder.yml:3]: Nice :+1: I think this is a really good combination; it avoids a new comment being posted with every commit.

File: .github/workflows/changelog_reminder.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    - name: Changelog Reminder
+      uses: peterjgrainger/action-changelog-reminder@v1.3.0
+      with:
+        changelog_regex: '.github/CHANGELOG.md'
+        customPrMessage: |
+          Beep! Boop! I see you're missing a changelog! 
+          Please edit [.github/CHANGELOG.md](/PennyLaneAI/pennylane/blob/master/.github/CHANGELOG.md) with:
+          * Description of the change
+          * Link back to this PR
+          * Your name in the contributors section
```
Comment (josh146) [inline .github/workflows/changelog_reminder.yml:20]:
```suggestion
          * A one-to-two sentence description of the change. You may include a
            small working example for new features.
          * A link back to this PR.
          * Your name (or GitHub username) in the contributors section.
```

File: .github/workflows/changelog_reminder.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    name: Changelog Reminder
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+    - name: Changelog Reminder
+      uses: peterjgrainger/action-changelog-reminder@v1.3.0
+      with:
+        changelog_regex: '.github/CHANGELOG.md'
+        customPrMessage: |
+          Beep! Boop! I see you're missing a changelog!
```
Comment (josh146) [inline .github/workflows/changelog_reminder.yml:16]: How do we feel about the Beep Boop? I know I suggested it, but I'm worried its too on-the-nose/cutesy 😆 

Alternatively, we could go with a Robot emoji?
```suggestion
         Hello. You may have forgotten to update the changelog! 🤖 
```

**Review by josh146 — COMMENTED** [review]
Great to finally have this in @albi3ro, just since monday I've caught 2 PRs without changelog updates! I also quite like how this action is set up to only comment on opened and ready for review PRs.

I've left a couple of suggestions re: the PR message. Once the wording is tightened, I think this one is good for merging.

File: .github/workflows/changelog_reminder.yml (hunk 3 above)
Comment (albi3ro) [reply .github/workflows/changelog_reminder.yml:16]: Not the biggest fan of the emoji, but otherwise the change is good

Comment (josh146) [reply .github/workflows/changelog_reminder.yml:16]: Okay, lets go for a version without the emoji:
```suggestion
          Hello. You may have forgotten to update the changelog!
```

File: .github/workflows/changelog_reminder.yml (hunk 1 above)
Comment (albi3ro) [reply .github/workflows/changelog_reminder.yml:3]: thanks

File: .github/workflows/changelog_reminder.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+    - name: Changelog Reminder
+      uses: peterjgrainger/action-changelog-reminder@v1.3.0
+      with:
+        changelog_regex: '.github/CHANGELOG.md'
+        customPrMessage: |
+          Hello. You may have forgotten to update the changelog!
+          Please edit [.github/CHANGELOG.md](/PennyLaneAI/pennylane/blob/master/.github/CHANGELOG.md) with:
```
Comment (antalszava) [inline .github/workflows/changelog_reminder.yml:17]: Would the complete URL be required here? The link that was posted as a comment doesn't seem to work.
```suggestion
          Please edit [.github/CHANGELOG.md](https://github.com/PennyLaneAI/pennylane/blob/master/.github/CHANGELOG.md) with:
```

File: .github/workflows/changelog_reminder.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
+on:
+  pull_request:
+    types: [opened, ready_for_review]
```
Comment (antalszava) [inline .github/workflows/changelog_reminder.yml:3]: How about also at `review_requested`?

**Review by antalszava — COMMENTED** [review]
This is looking awesome! :1st_place_medal: Will help a lot :slightly_smiling_face:

Would be curious to see the latest version of the comment be posted, could it be triggered? Also, seemed like the Python 3.8 CI check failed for some reason, retriggered it.

**Review by antalszava — APPROVED** [review]
Looks good to me! :100:

File: .github/workflows/changelog_reminder.yml (hunk 4 above)
Comment (albi3ro) [reply .github/workflows/changelog_reminder.yml:17]: I had the the link and text location flipped before. Works now.

File: .github/workflows/changelog_reminder.yml (hunk 5 above)
Comment (albi3ro) [reply .github/workflows/changelog_reminder.yml:3]: I assumed ready_for_review and review_requested would happen at about the same time, and I don't want it being repetitiveve

**Pull request merged**